File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        long cnt = 0;
        int l = 0, r = 0, n = s.length();
        long res = 0;
        while (r < n) {
            char high = s.charAt(r);
            char low = s.charAt(l);
            if ((r - l + 1) > k) {
                if (low == 'a' || low == 'e' || low == 'i' || low == 'o' || low == 'u') {
                    cnt--;
                }
                l++;
            }
            if ((r - l + 1) <= k) {
                if (high == 'a' || high == 'e' || high == 'i' || high == 'o' || high == 'u') {
                    cnt++;
                }
            }
            res = Math.max(res, cnt);
            r++;
        }
        return (int) res;
    }
}

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        l,r,cnt,res=0,0,0,0

        while r<len(s):
            if s[r] in {'a','e','i','o','u'}:
                cnt+=1
            while (r-l)>=k:
                if s[l] in  {'a','e','i','o','u'}:
                    cnt-=1
                l+=1
            res =max(res,cnt)
            r+=1
        return res            